src/spinner/src/Spinner.js

```diff
@@ -38,84 +38,63 @@ const innerClass = color =>
     fill: 'transparent'
   }).toString()
 
-class Spinner extends PureComponent {
-  static propTypes = {
-    /**
-     * Composes the Box component as the base.
-     */
-    ...Box.propTypes,
+function Spinner({ delay = 0, size = 40, ...props }, ref) {
+  const theme = useTheme()
+  const [isVisible, setIsVisible] = useState(delay === 0)
 
-    /**
-     * Delay after which spinner should be visible.
-     */
-    delay: PropTypes.number,
-
-    /**
-     * The size of the spinner.
-     */
-    size: PropTypes.number.isRequired,
-
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired
-  }
-
-  static defaultProps = {
-    size: 40,
-    delay: 0
-  }
-
-  constructor({ delay }) {
-    super()
-
-    this.state = {
-      isVisible: delay === 0
+  useEffect(() => {
+    let delayTimer = null
+    if (delay > 0) {
+      delayTimer = setTimeout(() => {
+        setIsVisible(true)
+      }, delay)
     }
-  }
 
-  render() {
-    if (!this.state.isVisible) {
-      return null
+    return function() {
+      clearTimeout(delayTimer)
     }
+  }, [])
 
-    const { theme, size, ...props } = this.props
-    return (
-      <Box width={size} height={size} lineHeight={0} {...props}>
-        <Box
-          is="svg"
-          className={outerClass}
-          x="0px"
-          y="0px"
-          viewBox="0 0 150 150"
-        >
-          <Box
-            is="circle"
-            className={innerClass(theme.spinnerColor)}
-            cx="75"
-            cy="75"
-            r="60"
-          />
-        </Box>
-      </Box>
-    )
+  if (!isVisible) {
+    return null
   }
 
-  componentDidMount() {
-    const { delay } = this.props
-
-    if (delay > 0) {
-      this.delayTimer = setTimeout(() => {
-        this.setState({
-          isVisible: true
-        })
-      }, delay)
-    }
-  }
+  return (
+    <Box width={size} height={size} lineHeight={0} {...props} innerRef={ref}>
+      <Box
+        is="svg"
+        className={outerClass}
+        x="0px"
+        y="0px"
+        viewBox="0 0 150 150"
+      >
+        <Box
+          is="circle"
+          className={innerClass(theme.spinnerColor)}
+          cx="75"
+          cy="75"
+          r="60"
+        />
+      </Box>
+    </Box>
+  )
+}
 
-  componentWillUnmount() {
-    clearTimeout(this.delayTimer)
-  }
+Spinner.propTypes = {
```

Comment:
Apparently we have to add propTypes to the final resulting exported component instead.